src/H5Gloc.c

```diff
@@ -410,9 +410,12 @@ H5G_loc_find(const H5G_loc_t *loc, const char *name, H5G_loc_t *obj_loc /*out*/)
 
     /* Check args. */
     assert(loc);
-    assert(name && *name);
```

Comment:
This was failing in debug builds due to the assertion on `*name`, but would pass for release/production builds. That said, I don't think it's allowed to have an empty link name for an object, so I converted this to a regular error check.